azerothcore/azerothcore-wotlk
# fix(Core/Game): Rest & Inn Improvements

## CHANGES PROPOSED:
-  Implemented missing Rest Flag Masks
-  Improved rest flag checks for Inns, Cities & Sanctuaries
-  Remove nasty hack fixes that just made no sense
-  Attempt to improve area trigger radius checks

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3712




## TESTS PERFORMED:
- Watch the video here by @Maurowin: https://youtu.be/BwmdouFTv9Y




## HOW TO TEST THE CHANGES:
- Login a newly created character or pre-existing character (race doesn't matter)
- Head to any Inn, City or Sanctuary zone
- Check if player gets rest flag on entering the area
- Leave the zone and see if flag is updated




## KNOWN ISSUES AND TODO LIST:
- Radius around Inns are to large and require further updates are mentioned below in the feedback.



## Target branch(es):

## Feedback
To fix radius issues around Inns, more updates are required on Position functions across the core. Math for it is incorrect and most of the time does nothing. Another PR will need to be opened for that separately. Changes made thus far are working as intended according to @Maurowin and the test they have done.







 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
